src/Workspaces/Core/Portable/Classification/AbstractClassificationService.cs

```diff
@@ -88,7 +125,7 @@ public async Task AddSyntacticClassificationsAsync(Document document, TextSpan t
             temp.Free();
         }
 
-        protected void AddRange(ArrayBuilder<ClassifiedSpan> temp, List<ClassifiedSpan> result)
+        protected static void AddRange(ArrayBuilder<ClassifiedSpan> temp, List<ClassifiedSpan> result)
```

Comment:
>protected static void AddRange(ArrayBuilder<ClassifiedSpan> temp, List<ClassifiedSpan> result) [](start = 8, length = 94)

This method shouldn't be necessary. ArrayBuilder has AddRange

Reply:
ArrayBuilder.AddRange takes in an IEnumerable.  That can't be iterated without boxing.  Something we try to avoid heavily in classification due to how often it is called.

Reply:
I don't think that's true. It directly calls `ImmutableArray<T>.Builder.AddRange`, which is fined-tuned to allocate as little as possible.

Reply:
Actually, I'm reading it backwards. The order of the parameters is confusing, we actually calling `List<T>.AddRange`. But the same thing. I don't see the boxing.

Reply:
> But the same thing. I don't see the boxing.

if you call `List<T>.AddRange(arrayBuilder)` that boxes since `List<T>` will call into `IEnumerble<T>.GetEnumerator()` (producing a boxed `IEnumerator<T>`).  

In the approach we have here, we directly iterate over the `ArrayBuilder<T>` which doesn't box since it has a struct enumerator.

Reply:
Oh, I see what you mean. That's not boxing though :). It's allocating an enumerator.

Reply:
Never mind. we do box the enumerator...

Reply:
Would be useful to add a comment.